python/perf-kernels/streamk/03-matrix-multiplication-stream-k.py

```diff
@@ -0,0 +1,216 @@
+            print(f"{total_full_tiles_streamk=}")
+            print(f"{iters_per_tile=}")
+            print(f"{total_iters_streamk=}")
+            print("total_remainder_iters_streamk=", total_partial_tiles_streamk)
+        use_bias = False
+        # compute grid (work to do per SM on the first wave)
+        grids = total_programs_streamk
+        stride_bias = bias.stride(0) if use_bias else 0
+        #  P=P*0.0
+        #  locks=locks*0
+        kk = streamk_gemm[(grids, )](
+            a,
+            b,
+            c,
+            bias,
+            P,
+            locks,
+            M,
+            N,
+            K,
+            a.stride(0),
+            a.stride(1),
+            b.stride(0),
+            b.stride(1),
+            c.stride(0),
+            c.stride(1),
+            stride_bias,
+            BLOCK_SIZE_M=BLK_M,
+            BLOCK_SIZE_N=BLK_N,
+            BLOCK_SIZE_K=BLK_K,
+            GROUP_SIZE_M=gsize_m,
+            NUM_SMS=total_programs_streamk,
+            BIAS=use_bias,
+            EVEN_K=even_k,
+            num_stages=num_stages,
+            num_warps=num_warps,
+            waves_per_eu=waves_per_eu,
+            matrix_instr_nonkdim=mfmaInstrSize,
+            kpack=kpack,
+        )
+        if matmul._debug:
+            print(f"{kk.n_regs} registers used, {kk.n_spills} spills")
+
+    #     print(kk.asm['ttgir'])
+    #     print(kk.asm['amdgcn'])
+
+        return c
+
+    @staticmethod
+    def forward(ctx, a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, bias: torch.Tensor, P: torch.Tensor,
+                locks: torch.Tensor, grid: int, BLK_M=128, BLK_N=128, BLK_K=32, gsize_m=1, two_tiles=True, num_stages=3,
+                num_warps=4, waves_per_eu=2, mfmaInstrSize=16, kpack=1):
+        matmul._call(a=a, b=b, c=c, bias=bias, P=P, locks=locks, total_programs_streamk=grid, BLK_M=BLK_M, BLK_N=BLK_N,
+                     BLK_K=BLK_K, gsize_m=gsize_m, two_tiles=two_tiles, num_warps=num_warps, num_stages=num_stages,
+                     waves_per_eu=waves_per_eu, mfmaInstrSize=mfmaInstrSize, kpack=kpack)
+        return c
+
+
+# ---------------------------------------------------------------------------
+# Example and Benchmark
+# ---------------------------------------------------------------------------
+
+perf = lambda ms: 2 * m * n * k * 1e-12 / (ms * 1e-3)
+
+#m, n, k = 4864, 4096, 8256  # some problem size to test
+#m, n, k = 4096, 4096, 8192  # some problem size to test
+#m, n, k = 1, 1024, 256
+#m, n, k = 8133, 8132, 8172  # some problem size to test
+#m, n, k = 8192, 8192, 8192  # some problem size to test
+#m, n, k = 8128, 6878, 7378  # some problem size to test
+#m, n, k = 8192, 4864, 6878  # some problem size to test
+#m, n, k = 512, 512, 512  # some problem size to test
+#m, n, k = 6912, 768, 256  # some problem size to test
+#m, n, k =4864, 8192, 4160  # some problem size to test
+#m, n, k = 5632, 6656, 7936
+m, n, k = 4864, 4096, 4300
```

Comment:
Can we maybe create a list of "interesting m, n, k" and just have those listed here and why. For example, one of these is a golden m, n, k that divides perfectly onto the 304 CUs and the tile size.

Reply:
done, see ea7b4050eea8532f3